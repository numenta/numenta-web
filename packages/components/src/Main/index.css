/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.main {
  composes: bg-white from 'tachyons';
  composes: br--bottom from 'tachyons';
  composes: br2 from 'tachyons';
  composes: center from 'tachyons';
  composes: mt5 from 'tachyons';
  composes: mw-100 from 'tachyons';
  composes: mw7-m from 'tachyons';
  composes: mw8-l from 'tachyons';
  composes: overflow-hidden from 'tachyons';
  composes: pb4 from 'tachyons';
  composes: ph3 from 'tachyons';
  composes: ph4-l from 'tachyons';
  composes: pt3 from 'tachyons';
  composes: shadow-4 from 'tachyons';
}
