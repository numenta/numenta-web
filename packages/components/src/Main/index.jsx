// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


/**
 *
 */
const Main = ({children}) => (
  <main className={styles.main}>
    {children}
  </main>
)

Main.propTypes = {
  children: React.PropTypes.node.isRequired,
}

export default Main
