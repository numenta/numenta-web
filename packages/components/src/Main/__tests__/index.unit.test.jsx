import React from 'react'
import renderer from 'react-test-renderer'

import Main from '../../Main'


describe('Main React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Main>
        <div>Hello</div>
      </Main>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
