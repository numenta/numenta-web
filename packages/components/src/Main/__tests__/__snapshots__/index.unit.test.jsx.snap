exports[`Main React component Renders correctly 1`] = `
<main
  className="main">
  <div>
    Hello
  </div>
</main>
`;
